cfme/cloud/provider.py

```diff
@@ -157,6 +170,27 @@ def create(self, *args, **kwargs):
                 infra_provider_name = str(self.infra_provider)
             setup_provider_by_name(
                 infra_provider_name, validate=True, check_existing=True)
+
+    @variable(alias='rest')
+    def create(self, *args, **kwargs):
+        self._check_infra()
+        if self.infra_provider:
+            # Do REST lookup for ID
+            # Run the super create
+            if isinstance(self.infra_provider, OpenstackInfraProvider):
+                infra_provider_name = self.infra_provider.name
+            else:
+                infra_provider_name = str(self.infra_provider)
+            provider_sr = rest_api().collections.providers.find_by(name=infra_provider_name)
+            if provider_sr.count > 0:
+                p_id = provider_sr[0].provider_id
+            self.rest_infra_provider = p_id
```

Comment:
It will be better to use  .get() which works similarly to find_by.
provider = rest_api().collections.providers.get(name=infra_provider_name)
self.rest_infra_provider = provider.id